sonic_installer/bootloader/aboot.py

```diff
@@ -179,3 +182,25 @@ def base64Decode(cls, text):
     def detect(cls):
         with open('/proc/cmdline') as f:
             return 'Aboot=' in f.read()
+
+    def _get_swi_file_offset(self, swipath, filename):
+        with zipfile.ZipFile(swipath) as swi:
+            with swi.open(filename) as f:
+                return f._fileobj.tell() # pylint: disable=protected-access
+
+    @contextmanager
+    def get_rootfs_path(self, image_path):
+        rootfs_path = os.path.join(image_path, ROOTFS_NAME)
+        if os.path.exists(rootfs_path):
```

Comment:
>if os.path.exists(rootfs_path): [](start = 8, length = 31)

Could you just check isSecureboot() ?

Reply:
I could have but there are some future use cases where `isSecureboot` will have to be changed to what I wrote in this review. It was a conscious decision.

Reply:
My point is that the path exists is not the ground truth for secure boot. So if you need to check secure boot, just check secure boot.
Let's assume under secure boot, someone intentionally or unintentionally create rootfs_path folder, is your logic happy?

---
In reply to: [562283015](https://github.com/Azure/sonic-utilities/pull/1366#discussion_r562283015) [](ancestors = 562283015)

Reply:
I see your point. I'll change this codepath to use `isSecureboot()`